Userland/Applications/FileManager/main.cpp

```diff
@@ -187,6 +189,58 @@ void do_create_link(Vector<String> const& selected_file_paths, GUI::Window* wind
     }
 }
 
+void do_create_archive(Vector<String> const& selected_file_paths, GUI::Window* window)
+{
+    String archive_name;
+    if (GUI::InputBox::show(window, archive_name, "Enter name:", "Create Archive") != GUI::InputBox::ExecOK)
+        return;
+
+    auto output_directory_path = LexicalPath(selected_file_paths.first());
+
+    StringBuilder path_builder;
+    path_builder.append(output_directory_path.dirname());
+    path_builder.append("/");
+    if (archive_name.is_empty()) {
+        path_builder.append(output_directory_path.parent().basename());
+        path_builder.append(".zip");
+    } else {
+        path_builder.append(archive_name);
+        if (!archive_name.ends_with(".zip"))
+            path_builder.append(".zip");
+    }
+    String output_path = path_builder.build();
+
+    pid_t zip_pid = fork();
+    if (zip_pid < 0) {
+        perror("fork");
+        VERIFY_NOT_REACHED();
+    }
+
+    if (!zip_pid) {
+        Vector<String> relative_paths;
+        Vector<char const*> arg_list;
+        arg_list.append("/bin/zip");
+        arg_list.append("-r");
+        arg_list.append("-f");
+        arg_list.append(output_path.characters());
+        for (auto const& path : selected_file_paths) {
+            relative_paths.append(LexicalPath::relative_path(path, output_directory_path.dirname()));
+            arg_list.append(relative_paths.last().characters());
+        }
+        arg_list.append(nullptr);
+        int rc = execvp("/bin/zip", const_cast<char* const*>(arg_list.data()));
```

Comment:
isn't that `char const**`?

Reply:
I think this is right, the signature is `char* const argv[]`